test/unittests/test_xmlupload/test_stash_circular_references.py

```diff
@@ -0,0 +1,19 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+import pytest
+
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
+from dsp_tools.utils.xmlupload.stash_circular_references import remove_circular_references
+from dsp_tools.utils.xmlupload.xmlupload import _extract_resources_from_xml
+
+test_root = parse_and_clean_xml_file("testdata/generate_test_data/test_circular_references_1.xml")
+resources = _extract_resources_from_xml(test_root, "simcir")
+
+
+def test_remove_circular_references() -> None:
```

Comment:
I agree with Balduin, the naming of this file + function should be more telling

Reply:
I named this file after the original in a generic way so that further tests could be added in the future. But if you would like to handle that differently, I am fine with that.

Reply:
IMHO, names should always be as precise as possible. And if we add something in the future, we can still rename it then.